packages/commonwealth/client/scripts/controllers/app/webWallets/metamask_web_wallet.ts

```diff
@@ -50,10 +61,13 @@ class MetamaskWebWalletController implements IWebWallet<string> {
     return signature;
   }
 
-  public async signLoginToken(message: string): Promise<string> {
-    const msgParams = constructTypedMessage(
+  public async signLoginToken(validationBlockInfo: string): Promise<string> {
+    const sessionPublicAddress = app.sessions.getOrCreateAddress(app.chain?.meta.node.ethChainId || 1);
+    const msgParams = await constructTypedMessage(
+      this.accounts[0],
```

Comment:
why are we using `this.accounts[0]` instead of the account that was passed into `signWithAccount`? can there ever be more than one account?

Reply:
Yeah, but I'm not sure how different web wallets handle this - Metamask lets you connect multiple accounts to a site, but usually people only connect one, as opposed to all of their accounts.

Reply:
It looks like in the wallets list code https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/components/component_kit/cw_wallets_list.tsx#L310 it actually forces the user to log in with the first account.

You can see that it always uses the first account if you switch to the 2nd account in your wallet (e.g. MetaMask) and try to log in. There is a warning tooltip that tells you that you are trying to sign a message using an account that is not the currently selected one. Is this really the behaviour we want? What if a user really does want to log into Commonwealth using a different account?

<img width="348" alt="Screenshot 2022-10-27 at 3 54 44 pm" src="https://user-images.githubusercontent.com/457206/198324087-773cba72-6c0b-4b38-95d0-f1901f6b6cc2.png">

The tooltip:
<img width="354" alt="Screenshot 2022-10-27 at 3 54 48 pm" src="https://user-images.githubusercontent.com/457206/198324022-a9b626d5-63c6-44a7-bda4-ddad0c5a4a6e.png">

Reply:
Is there a way to get the currently selected account from `ethers`/metamask? If so, we should implement that though in a separate PR. A lot of apps I've used don't seem to handle this case but it's definitely better UX to respect the user's currently selected account.